src/AssimpWorker/import/ColladaImporter/ColladaRecursiveImporter.cpp

```diff
@@ -0,0 +1,156 @@
+/*
+* This file is part of ATLAS. It is subject to the license terms in
+* the LICENSE file found in the top-level directory of this distribution.
+* (Also avialable at http://www.apache.org/licenses/LICENSE-2.0.txt)
+* You may not use this file except in compliance with the License.
+*/
+
+#include <iostream>
+#include "ColladaRecursiveImporter.hpp"
+#include "../../internal/Exception.hpp"
+
+namespace AssimpWorker {
+
+	using ATLAS::Model::DataDeletingBlob;
+	using ATLAS::Model::Folder;
+
+	ColladaRecursiveImporter::ColladaRecursiveImporter(const Poco::URI& colladaFileURI, Log& log, const std::string& pathToWorkingDirectory, ColladaMassagerRegistry& registry, float scale) :
+		Importer(colladaFileURI.toString(), log),
+		pathToWorkingDirectory(pathToWorkingDirectory),
+		colladaFileURI(colladaFileURI),
+		childImporter(),
+		importer(nullptr),
+		massagerRegistry(registry),
+		massager(nullptr),
+		parentScale(scale)
+	{
+		return;
+	}
+
+	ColladaRecursiveImporter::~ColladaRecursiveImporter(){
+		delete importer;
+		for (auto ci : childImporter){
+			delete ci;
+		}
+	}
+
+	void ColladaRecursiveImporter::addElementsTo(Folder& root){
+		preprocessCollada();
+		const aiScene* scene = runAssimpImport();
+		convertToFolderStructure(scene, root);
+		importChildColladas(root);
+	}
+
+	void ColladaRecursiveImporter::preprocessCollada(){
+		massager = massagerRegistry.getMassager(colladaFileURI);
+		massager->massage();
+		if (parentScale != -1 && massager->getCurrentUnit() != parentScale) {
+			throw Exception("Inconsistent scales used in input files.");
+		}
+	}
+
+	const aiScene* ColladaRecursiveImporter::runAssimpImport(){
+		this->importer = new AssimpWorker::AssimpImporter();
+		const aiScene* scene = importer->importSceneFromFile(colladaFileURI.getPath(), log);
+		if (!scene) {
+			throw Exception("Error while running Assimp.");
+		}
+		return scene;
+	}
+
+	void ColladaRecursiveImporter::convertToFolderStructure(const aiScene* scene, Folder& root){
+		const std::string fragment = colladaFileURI.getFragment();
+		if (fragment != "") {
+			aiNode* startingPointToImportFrom = findaiNodeWithName(scene->mRootNode, fragment);
+			if (startingPointToImportFrom == nullptr){
+				throw Exception("Could not find a Node with id '" + fragment + "'");
```

Comment:
This (and the following) error message is (almost) the same, although for different errors. Could be a little more informative.